### PR TITLE
Add QR Code "Scan to share" button that dispatches QR payload

### PR DESCRIPTION
**Pull request by Joey-Silberman**
This PR adds a "Scan to share" button to the wallet home view (if there are any `rawCredentials` on the screen) which allows the user to scan a QR code whose payload will be dispatched with the `dispatchQRcode` function which is not yet implemented. The button will not appear if there are no credentials added yet.

I also bumped the CompileSDK version to match the newest version of Android Studio.

Note: This is the Android version of this PR: https://github.com/spruceid/mobile-sdk-ios-app/pull/21

### CODE REVIEW

File: example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt [hunk 1]
```diff
@@ -86,33 +94,73 @@ fun WalletHomeHeader(navController: NavController) {
...
+                            contentDescription = "QR Code Icon", // Accessibility description
+                            tint = Color.White, // Icon color
+                            modifier = Modifier.padding(end = 10.dp) // Space between icon and text
+                        )
+                        Text(
+                            text = "Scan to share",
+                            fontFamily = Inter, // Ensure you have the Inter font available
+                            fontWeight = FontWeight.Normal,
+                            fontSize = 15.sp,
+                        )
```
Comment (Ryanmtate) [inline example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt:160]: remove comments.

**Review by Ryanmtate — APPROVED** [review]

File: example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt [hunk 2]
```diff
@@ -86,33 +94,72 @@ fun WalletHomeHeader(navController: NavController) {
...
+    //        }
+            }
+
+            Box(
+                modifier = Modifier.fillMaxSize(),
+                contentAlignment = Alignment.BottomCenter
+            ) {
+                Button(
+                    onClick = {
+                        navController.navigate(Screen.OID4VPScreen.route)
```
Comment (Ryanmtate) [inline example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt:136]: In a follow-up PR, we need to implement the QR code scanner on this page route.

File: example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt [hunk 3]
```diff
@@ -86,33 +94,72 @@ fun WalletHomeHeader(navController: NavController) {
...
-                .fillMaxWidth()
-                .padding(top = 20.dp)
-        ) {
-            items(rawCredentials) { rawCredential ->
-                AchievementCredentialItem(
-                    rawCredential.rawCredential,
-                    onDelete = {
-                        scope.launch {
-                            rawCredentialsViewModel.deleteRawCredential(id = rawCredential.id)
+    if(!rawCredentials.isNotEmpty()) {
```
Comment (Juliano1612) [inline example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt:105]:
```suggestion
    if(rawCredentials.isNotEmpty()) {
```

File: example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt [hunk 4]
```diff
@@ -86,33 +94,72 @@ fun WalletHomeHeader(navController: NavController) {
...
+                    Row(
+                        verticalAlignment = Alignment.CenterVertically,
+                        modifier = Modifier.padding(8.dp)
+                    ) {
+                        Icon(
+                            painter = painterResource(id = R.drawable.scan_qr_code),
+                            contentDescription = "QR Code Icon",
+                            tint = Color.White,
+                            modifier = Modifier.padding(end = 10.dp)
+                        )
```
Comment (Juliano1612) [inline example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt:153]: This icon is gray, but should be white

File: example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt [hunk 5]
```diff
@@ -86,33 +94,72 @@ fun WalletHomeHeader(navController: NavController) {
...
+                modifier = Modifier.fillMaxSize(),
+                contentAlignment = Alignment.BottomCenter
+            ) {
+                Button(
+                    onClick = {
+                        navController.navigate(Screen.OID4VPScreen.route)
+                    },
+                    modifier = Modifier.fillMaxWidth(),
+                    colors = ButtonDefaults.buttonColors(
+                        containerColor = Color.Blue,
```
Comment (Juliano1612) [inline example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt:140]: This is not the color from the designs

File: example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt [hunk 6]
```diff
@@ -86,33 +94,72 @@ fun WalletHomeHeader(navController: NavController) {
...
+    //            }
+    //            ShareableCredentialListItems(mdocBase64 = mdocBase64)
+    //        }
+            }
+
+            Box(
+                modifier = Modifier.fillMaxSize(),
+                contentAlignment = Alignment.BottomCenter
+            ) {
+                Button(
```
Comment (Juliano1612) [inline example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt:134]: The button is not fixed on the screen. If I add five credentials it disappears

**Review by Juliano1612 — CHANGES_REQUESTED** [review]
The button and the redirection are working well, but there are some design issues

File: example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt (hunk 6 above)
Comment (Joey-Silberman) [reply example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt:134]: I think the button should be fixed on the screen now, let me know if it works on your end.

**Review by timothee-haudebourg — CHANGES_REQUESTED** [review]
Seems like it's almost ready! I just have the same comment as for the Android PR: this PR should have no mention of OID4VP. The task is to add a QR-code button handing over the QR-code payload to a generic dispatch function.

It is correct that in the near future we will extend the dispatch function to trigger the OID4VP flow, but this PR should be completely agnostic about that.

**Comment by timothee-haudebourg** [comment]
Question: what happens currently when the QR-code is scanned?

File: example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt [hunk 7]
```diff
@@ -86,33 +94,72 @@ fun WalletHomeHeader(navController: NavController) {
...
+                    )
+                ) {
+                    Row(
+                        verticalAlignment = Alignment.CenterVertically,
+                        modifier = Modifier.padding(8.dp)
+                    ) {
+                        Icon(
+                            painter = painterResource(id = R.drawable.scan_qr_code_white),
+                            contentDescription = "QR Code Icon",
+                            tint = Color(0xFFFFFFFF),
```
Comment (Juliano1612) [inline example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt:151]: you can use `Color.White`

File: example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt [hunk 8]
```diff
@@ -86,33 +94,72 @@ fun WalletHomeHeader(navController: NavController) {
...
+                modifier = Modifier.fillMaxSize(),
+                contentAlignment = Alignment.BottomCenter
+            ) {
+                Button(
+                    onClick = {
+                        navController.navigate(Screen.OID4VPScreen.route)
+                    },
+                    modifier = Modifier.fillMaxWidth(),
+                    colors = ButtonDefaults.buttonColors(
+                        containerColor = Color(0xFF488CF4),
```
Comment (Juliano1612) [inline example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt:140]: you must create a color variable, the same way we have for `CTAButtonGreen`. You can call it `CTAButtonBlue` and create it on `Color.kt` and `colors.xml`

File: example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt [hunk 9]
```diff
@@ -86,33 +94,72 @@ fun WalletHomeHeader(navController: NavController) {
...
-                AchievementCredentialItem(
-                    rawCredential.rawCredential,
-                    onDelete = {
-                        scope.launch {
-                            rawCredentialsViewModel.deleteRawCredential(id = rawCredential.id)
+        Box(modifier = Modifier.fillMaxSize()) {
+            LazyColumn(
+                Modifier
+                    .fillMaxWidth()
+                    .padding(top = 20.dp)
```
Comment (Juliano1612) [inline example/src/main/java/com/spruceid/mobilesdkexample/wallet/WalletHomeView.kt:110]:
```suggestion
                    .padding(top = 20.dp)
                    .padding(bottom = 60.dp)
```

**Review by Juliano1612 — CHANGES_REQUESTED** [review]
LGTM! The button is working and is always visible! Just a few code suggestions to finish

**Comment by Juliano1612** [comment]
> Question: what happens currently when the QR-code is scanned?

It only redirects to a new component that will handle the OID4VP integration, but the OID4VP flow will be integrated in another PR

**Review by Juliano1612 — APPROVED** [review]

**Comment by Ryanmtate** [comment]
> > Question: what happens currently when the QR-code is scanned?
> 
> It only redirects to a new component that will handle the OID4VP integration, but the OID4VP flow will be integrated in another PR

https://github.com/spruceid/mobile-sdk-kt/pull/37 contains a handler to navigate to an OID4VP view.

**Review by timothee-haudebourg — APPROVED** [review]

**Pull request merged**